casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -258,43 +260,54 @@ object ProtoUtil {
     } yield ps.blockNumber).getOrElse(0L)
 
   /*
-   * Two blocks conflict if they both use the same deploy in different histories
-   *
-   * TODO: Update the logic of this function to make use of the trace logs and
-   * say that two blocks don't conflict if they act on disjoint sets of channels
+   * Block b1 conflicts with b2 if any of b1's ancestors contains a replay log entry that
+   * touches a channel that any of b2's ancestors' (that are not common with b1's ancestors)
+   * replay log entries touch.
    */
   def conflicts[F[_]: Monad: BlockStore](
       b1: BlockMessage,
       b2: BlockMessage,
       genesis: BlockMessage,
       dag: BlockDag
-  ): F[Boolean] =
+  ): F[Boolean] = {
+    val b1MetaData       = dag.dataLookup.get(b1.blockHash)
+    val b2MetaData       = dag.dataLookup.get(b2.blockHash)
+    val blockMetaDataSeq = Vector(b1MetaData, b2MetaData).flatten
+    if (blockMetaDataSeq.size == 2) {
```

Comment:
I would be for raising an error when one (or both) of the blocks isn't present in the DAG. Otherwise the program might fail elsewhere with a completely unrelated reason